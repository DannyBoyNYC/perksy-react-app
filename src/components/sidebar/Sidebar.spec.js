const React = require('react');

const Sidebar = require('./index').default;

describe('<Sidebar.* />', () => {
  let component;
  const requiredProps = {
    links: [
      {
        text: 'Foo',
        url: '#'
      },
      {
        text: 'Bar',
        url: '#'
      }
    ],
    user: {}
  };

  describe('<Sidebar.Static />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Sidebar.Static {...requiredProps} />,
        'Sidebar'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a Logo.White component', () => {
      expect(component).toContainExactlyOneMatchingElement('White');
    });

    it('does not render a Button.Info', () => {
      expect(component.find('Info')).not.toExist();
    });

    it('renders utility links if they are given', () => {
      component.setProps({
        ultilityLinks: [{ text: 'utility', url: '/link' }]
      });
      expect(component).toContainMatchingElements(2, 'Vertical');
    });
  });

  describe('<Sidebar.Collapsible />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Sidebar.Collapsible {...requiredProps} />,
        'Sidebar'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('renders a Button.Info to toggle state', () => {
      expect(component.find('Info')).toExist();
    });

    it('adds a `.collapsed` class when the component is collapsed', () => {
      component.setState({
        isCollapsed: true
      });
      expect(component).toHaveClassName('collapsed');
    });

    it('will toggle isCollapsed when Button.Info is clicked', () => {
      component.find('Info').simulate('click');
      expect(component).toHaveClassName('collapsed');
      expect(component).toHaveState({ isCollapsed: true });
      component.find('Info').simulate('click');
      expect(component).not.toHaveClassName('collapsed');
      expect(component).toHaveState({ isCollapsed: false });
    });
  });
});
