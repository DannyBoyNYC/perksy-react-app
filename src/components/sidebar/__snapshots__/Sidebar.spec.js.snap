// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar.* /> <Sidebar.Collapsible /> should render correctly with all required props 1`] = `
<aside
  className="Sidebar-root-0-1-1"
>
  <Info
    className="Sidebar-toggle-0-1-2"
    onClick={[Function]}
  >
    X
  </Info>
  <White
    className="Sidebar-logo-0-1-3"
  />
  <Avatar
    className="Sidebar-user-0-1-4"
  />
  <Vertical
    className="Sidebar-navigation-0-1-5"
    links={
      Array [
        Object {
          "text": "Foo",
          "url": "#",
        },
        Object {
          "text": "Bar",
          "url": "#",
        },
      ]
    }
  />
</aside>
`;

exports[`<Sidebar.* /> <Sidebar.Static /> should render correctly with all required props 1`] = `
<aside
  className="Sidebar-root-0-1-1 Sidebar-static-0-1-7"
>
  <White
    className="Sidebar-logo-0-1-3"
  />
  <Avatar
    className="Sidebar-user-0-1-4"
  />
  <Vertical
    className="Sidebar-navigation-0-1-5"
    links={
      Array [
        Object {
          "text": "Foo",
          "url": "#",
        },
        Object {
          "text": "Bar",
          "url": "#",
        },
      ]
    }
  />
</aside>
`;
